instrumentation/aws-sdk/aws-sdk-1.11/src/main/java/io/opentelemetry/auto/instrumentation/awssdk/v1_11/TracingRequestHandler.java

```diff
@@ -49,38 +45,29 @@ public AmazonWebServiceRequest beforeMarshalling(final AmazonWebServiceRequest r
 
   @Override
   public void beforeRequest(final Request<?> request) {
-    Span span = decorate.getOrCreateSpan(request, TRACER);
-    decorate.afterStart(span);
-    decorate.onRequest(span, request);
+    Span span = tracer.startSpan(request);
     request.addHandlerContext(
         SPAN_SCOPE_PAIR_CONTEXT_KEY,
-        new SpanWithScope(
-            span, ContextUtils.withScopedContext(ClientDecorator.currentContextWith(span))));
+        new SpanWithScope(span, withScopedContext(currentContextWith(span))));
```

Comment:
I think this logic should be in `tracer.startScope`

Reply:
This has been refactored in #922.